sfcli/SmartFace.Cli/Core/Domain/WatchlistMember/Impl/WatchlistMemberRegistrationManager.cs

```diff
@@ -189,37 +199,55 @@ private string GetFullPathToRegistrationDirectory(string directory)
             return registrationDir;
         }
 
-        private IEnumerable<WatchlistMemberRegisterData> BuildWatchlistMemberRegisterData(string registrationDirectory, string[] watchlistIds, RegisterRequestParams requestParams)
+        private IEnumerable<WatchlistMemberRegisterData> BuildWatchlistMemberRegisterData(
+            string registrationDirectory,
+            string fileNameToProperty,
+            string[] watchlistIds,
+            RegisterRequestParams requestParams)
         {
             var files = Directory.GetFiles(registrationDirectory);
 
             var watchlistMemberPhotoPaths = new List<WatchlistMemberPhotoPath>();
 
             foreach (var file in files)
             {
-                if (!TryGetWatchlistMemberIdFromFile(file, out var watchlistMemberId))
+                if (!TryGetWatchlistMemberIdFromFile(file, out var validFileName))
                 {
                     continue;
                 }
 
-                var watchlistMemberPhotoPath = new WatchlistMemberPhotoPath(watchlistMemberId, file);
-                watchlistMemberPhotoPaths.Add(watchlistMemberPhotoPath);
+                switch (fileNameToProperty.ToLower())
+                {
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(fileNameToProperty));
+
+                    case "id":
```

Comment:
Yes, this could be Enum, however I am not sure if Command line lib works well with Enums

Reply:
I'm not sure either, but you could test it :) Either that or migrate to different command line lib that supports enums, e.g. `System.CommandLine`